patches/0010-allow-TLS-1.3-when-crypto-tls-fipsonly-is-imported.patch

```diff
@@ -0,0 +1,128 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Quim Muntal <qmuntaldiaz@microsoft.com>
+Date: Mon, 16 Oct 2023 12:04:32 +0000
+Subject: [PATCH] allow TLS 1.3 when crypto/tls/fipsonly is imported
+
+---
+ src/crypto/tls/boring.go                 | 2 +-
+ src/crypto/tls/boring_test.go            | 8 +++++---
+ src/crypto/tls/cipher_suites.go          | 5 +++++
+ src/crypto/tls/handshake_client.go       | 4 +++-
+ src/crypto/tls/handshake_client_tls13.go | 4 ----
+ src/crypto/tls/handshake_server_tls13.go | 8 +++-----
+ 6 files changed, 17 insertions(+), 14 deletions(-)
+
+diff --git a/src/crypto/tls/boring.go b/src/crypto/tls/boring.go
+index ecd0f5a7b3e9ed..38d546a607774f 100644
+--- a/src/crypto/tls/boring.go
++++ b/src/crypto/tls/boring.go
+@@ -24,7 +24,7 @@ func fipsMinVersion(c *Config) uint16 {
+ // fipsMaxVersion replaces c.maxVersion in FIPS-only mode.
+ func fipsMaxVersion(c *Config) uint16 {
+ 	// FIPS requires TLS 1.2.
+-	return VersionTLS12
++	return VersionTLS13
+ }
+ 
+ // default defaultFIPSCurvePreferences is the FIPS-allowed curves,
+diff --git a/src/crypto/tls/boring_test.go b/src/crypto/tls/boring_test.go
+index 86595e588cf604..6c085351e7d33d 100644
+--- a/src/crypto/tls/boring_test.go
++++ b/src/crypto/tls/boring_test.go
+@@ -52,11 +52,11 @@ func TestBoringServerProtocolVersion(t *testing.T) {
+ 	test("VersionTLS10", VersionTLS10, "client offered only unsupported versions")
+ 	test("VersionTLS11", VersionTLS11, "client offered only unsupported versions")
+ 	test("VersionTLS12", VersionTLS12, "")
+-	test("VersionTLS13", VersionTLS13, "client offered only unsupported versions")
++	test("VersionTLS13", VersionTLS13, "")
+ }
+ 
+ func isBoringVersion(v uint16) bool {
+-	return v == VersionTLS12
++	return v == VersionTLS12 || v == VersionTLS13
+ }
+ 
+ func isBoringCipherSuite(id uint16) bool {
+@@ -66,7 +66,9 @@ func isBoringCipherSuite(id uint16) bool {
+ 		TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+ 		TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,
+ 		TLS_RSA_WITH_AES_128_GCM_SHA256,
+-		TLS_RSA_WITH_AES_256_GCM_SHA384:
++		TLS_RSA_WITH_AES_256_GCM_SHA384,
++		TLS_AES_128_GCM_SHA256,
++		TLS_AES_256_GCM_SHA384:
```

Comment:
Seeing TLS 1.3 suites bunched up with 1.2 suites, I wonder if it would make the tests more specific to separate out when 1.2 vs. 1.3 suites are expected. However: I don't know if this would actually catch anything, and it could be a significant chunk of work that never detects any problem and simply increases conflicts vs. upstream. I think we're better off waiting for upstream to reimplement this test with 1.3 in mind.

Reply:
Good point. I'll split it, this way `TestBoringServerCipherSuites` will also test that we are not returning TLS 1.3 ciphers in TLS 1.2 connections.